inspirehep/records/api/base.py

```diff
@@ -166,3 +179,228 @@ def hard_delete(self):
+            return bucket.bucket
+        return self._create_bucket(location, storage_class)
+
+    def _create_bucket(self, location=None, storage_class=None):
+        """Create bucket and return it.
+        Note:
+            Overwrites base_class._create_bucket method as it is not implemented
+            It can create more than one bucket for the same parameters.
+            It's private method, do not use it. Instead use `get_bucket()`
+
+        Args:
+            location (Location): Bucket location object
+                (default: 'RECORDS_DEFAULT_FILE_LOCATION_NAME') from config
+            storage_class (str): Bucket storage class
+                (default: 'RECORDS_DEFAULT_STORAGE_CLASS') from config
+
+        Returns: Bucket for current record, selected location and storage_class
+        """
+        bucket = Bucket.create(location=location, storage_class=storage_class)
+        RecordsBuckets.create(record=self.model, bucket=bucket)
+        return bucket
+
+    @property
+    def files(self):
+        """Get files iterator.
+        Original one was so ugly that it had to be changed!
+
+        Returns:
+            Iterator to files
+        """
+        if self.model is None:
+            raise MissingModelError()
+
+        bucket = self.get_bucket()
+
+        return self.files_iter_cls(self, bucket=bucket, file_cls=self.file_cls)
+
+    def _download_file_from_url(self, url):
+        """Downloads file and calculates hash for it
+        if everythong is ok then adds it to files in current record
+
+        Args:
+            url (str): Local or remote url/filepath
+
+        Returns (str): key(sha-1) of downloaded file
+
+        Example:
+            >>> self._download_file_from_url('http://example.com/url_to_file.pdf')
+            '207611e7bf8a83f0739bb2e16a1a7cf0d585fb5f'
+        """
+        stream = fsopen(url, mode="rb")
+        # TODO: change to stream.read() when fs will be updated to > 2.0
+        # As HTTPOpener is not working with size = -1
+        # (and read() method sets this size as default)
+        # This is workaround until we will update to fs >2.0
+        data = stream._f.wrapped_file.read()
+        key = hashlib.sha1(data).hexdigest()
+        if key not in self.files.keys:
+            self.files[key] = BytesIO(data)
+        return key
+
+    def _download_file_from_lcoal_storage(self, url, **kwargs):
+        """Opens local file with ObjectVersion API, callculates it's hash and returns
+        hash of the file
+        Args:
+            url (str): Local url which starts with /api/files/
+
+        Returns (str): key(sha-1) of downloaded file
+        Example:
+            >>> url = '/api/files/261926f6-4923-458e-adb0/207611e7bf8a83f0739bb2e'
+            >>> self._download_file_from_lcoal_storage(url)
+                '207611e7bf8a83f0739bb2e'
+        """
+        url_splited = url.split("/")
+        file = ObjectVersion.get(bucket=url_splited[-2], key=url_splited[-1])
```

Comment:
I'd move the url splitting in a separated, maybe ``static`` method, so we can easily test it.

Reply:
ok